rs/types/base_types/src/principal_id.rs

```diff
@@ -164,23 +165,46 @@ impl std::str::FromStr for PrincipalId {
     }
 }
 
-/// Some principal ids have special classes (system-generated,
-/// self-authenticating, derived), see <https://internetcomputer.org/docs/current/references/ic-interface-spec#id-classes>
-///
-/// The following functions allow creating and testing for the special forms.
+#[derive(Debug, Clone, Copy, PartialEq, Eq, EnumIter)]
+pub enum Class {
```

Comment:
I would call it PrincipalIdClass because otherwise it will be ambiguous everywhere you import it.